### PR TITLE
feat(themes): add type tokens to themes

### PR DESCRIPTION
**Pull request by joshblack**
Draft-only PR to show integration for theming for the theming project 👀

### CODE REVIEW

File: packages/components/src/globals/scss/vendor/@carbon/elements/scss/themes/_mixins.scss [hunk 1]
```diff
@@ -1,168 +1,30 @@
...
+  } @else {
+    --#{$prefix}-#{$name}: #{$value};
   }
+}
 
-  @content;
-  // Reset to default theme after apply in content
-  @if $theme != $carbon--theme {
-    @include carbon--theme();
+/// @access private
```
Comment (mattrosno) [inline packages/components/src/globals/scss/vendor/@carbon/elements/scss/themes/_mixins.scss:22]: Can you add `/// @group @carbon/themes` next to any new variable/mixin/function that's marked with private? Full SassDoc annotation being:

```scss
/// @access private
/// @group @carbon/themes
```

That way, these new items won't be grouped in the SassDoc "general" category.

File: packages/components/src/globals/scss/vendor/@carbon/themes/scss/_mixins.scss [hunk 2]
```diff
@@ -1,168 +1,30 @@
...
-    --highlight: #{map-get($theme, 'highlight')};
-    --skeleton-01: #{map-get($theme, 'skeleton-01')};
-    --skeleton-02: #{map-get($theme, 'skeleton-02')};
-    --brand-01: #{map-get($theme, 'brand-01')};
-    --brand-02: #{map-get($theme, 'brand-02')};
-    --brand-03: #{map-get($theme, 'brand-03')};
-    --active-01: #{map-get($theme, 'active-01')};
-    --hover-field: #{map-get($theme, 'hover-field')};
+/// @access private
+$custom-property-prefix: 'cds' !default !global;
```
Comment (mattrosno) [inline packages/components/src/globals/scss/vendor/@carbon/themes/scss/_mixins.scss:9]: Is `!global` needed here since it's not in a mixin?

**Review by mattrosno — COMMENTED** [review]
A couple suggestions here - did not dig too deeply in this initial review. I support breaking this up into smaller PRs such as:

- Add features to scss-generator
- Add features to themes package
- Add type tokens to themes (used as example for map support for tokens)

**Comment by joshblack** [comment]
@mattrosno yup! That's the plan (to break it down). First one is https://github.com/carbon-design-system/carbon/pull/4065

File: packages/components/src/globals/scss/vendor/@carbon/elements/scss/themes/_mixins.scss (hunk 1 above)
Comment (joshblack) [reply packages/components/src/globals/scss/vendor/@carbon/elements/scss/themes/_mixins.scss:22]: On it 👍

**Pull request closed without merging**